app/services/scheduled/service.go

```diff
@@ -0,0 +1,133 @@
+package scheduled
+
+import (
+	"github.com/hashgraph/hedera-sdk-go/v2"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/client"
+	hederahelper "github.com/limechain/hedera-eth-bridge-validator/app/helper/hedera"
+	"github.com/limechain/hedera-eth-bridge-validator/app/model/transfer"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/limechain/hedera-eth-bridge-validator/constants"
+	log "github.com/sirupsen/logrus"
+)
+
+type Service struct {
+	payerAccount     hedera.AccountID
+	hederaNodeClient client.HederaNode
+	mirrorNodeClient client.MirrorNode
+	logger           *log.Entry
+}
+
+func New(
+	payerAccount string,
+	hederaNodeClient client.HederaNode,
+	mirrorNodeClient client.MirrorNode) *Service {
+	payer, err := hedera.AccountIDFromString(payerAccount)
+	if err != nil {
+		log.Fatalf("Invalid payer account: [%s].", payerAccount)
+	}
+
+	return &Service{
+		payerAccount:     payer,
+		hederaNodeClient: hederaNodeClient,
+		mirrorNodeClient: mirrorNodeClient,
+		logger:           config.GetLoggerFor("Burn Event Service"),
+	}
+}
+
+// Execute submits a scheduled transaction and executes provided functions when necessary
+func (s *Service) Execute(
+	id, nativeAsset string,
+	transfers []transfer.Hedera,
+	onExecutionSuccess func(transactionID, scheduleID string), onExecutionFail, onSuccess, onFail func(transactionID string)) {
+	transactionResponse, err := s.executeScheduledTransaction(id, nativeAsset, transfers)
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to submit scheduled transaction. Error [%s].", id, err)
+		onExecutionFail(hederahelper.ToMirrorNodeTransactionID(transactionResponse.TransactionID.String()))
+		return
+	}
+
+	scheduledTxID := hederahelper.ToMirrorNodeTransactionID(transactionResponse.TransactionID.String())
+	s.logger.Infof("[%s] - Successfully submitted scheduled transaction [%s].",
+		id,
+		scheduledTxID)
+
+	txReceipt, err := transactionResponse.GetReceipt(s.hederaNodeClient.GetClient())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to get transaction receipt for [%s]", id, transactionResponse.TransactionID)
+		onExecutionFail(scheduledTxID)
+		return
+	}
+
+	switch txReceipt.Status {
+	case hedera.StatusIdenticalScheduleAlreadyCreated:
+		s.handleScheduleSign(id, *txReceipt.ScheduleID)
+	case hedera.StatusSuccess:
+		s.logger.Infof("[%s] - Updating db status to Submitted with TransactionID [%s].",
+			id,
+			hederahelper.ToMirrorNodeTransactionID(txReceipt.ScheduledTransactionID.String()))
```

Comment:
Do we need to log this here. Now that we have `onSuccess` and `onFail` it does not seem right for this function to know what we are going to do on success. It might be better to log it as `debug` in the `repository` `UpdateStatusSubmitted` as-well-as in the inner `updateStatus` function. Another way to do it is to log it in the definition of the callbacks in the `burn-event` service f.e